lib/srv/desktop/rdp/rdpclient/client.go

```diff
@@ -435,23 +449,25 @@ func (c *Client) handleRemoteCopy(data []byte) C.CGOErrCode {
 }
 
 // Wait blocks until the client disconnects and runs the cleanup.
-func (c *Client) Wait() error {
+func (c *Client) Wait() {
 	c.wg.Wait()
-	// Let the Rust side free its data.
-	C.free_rdp(c.rustClient)
-	return nil
+	c.Cleanup()
 }
 
-// Close shuts down the client and closes any existing connections.
-// It is safe to call multiple times, from multiple goroutines.
-// Calls other than the first one are no-ops.
-func (c *Client) Close() {
-	c.closeOnce.Do(func() {
+// Cleanup frees the memory of the cgo.Handle, closes the RDP client connection,
+// and frees the Rust client.
```

Comment:
This is an exported method, so you should write it from the perspective of an external package calling into the RDP client (meaning internal details like CGo handles and Rust clients should probably not be exposed).